plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/java/KSTypeReferenceJavaImpl.kt

```diff
@@ -50,7 +50,6 @@ class KSTypeReferenceJavaImpl(val psi: PsiType) : KSTypeReference {
         }
         when (type) {
             is PsiClassType -> KSClassifierReferenceJavaImpl.getCached(type)
-            is PsiMethodReferenceType -> KSCallableReferenceJavaImpl.getCached(type)
```

Comment:
what about references like `someObject::toString`?

Reply:
It does not have declaration site usage, only expression usage so method reference can't be a type. You can assign method reference to an object of SAM type, or pass it to function parameter of lambda types, where it is modeled in various functional interfaces(e.g. Predicate), too.